### PR TITLE
add pass config with imagePullPolicy feature.

### LINKED ISSUE #2444: Dapr sidecar docker image is always pulled from the registry (also when it is already present on the machine)
## In what area(s)?

/area-k8sinjector

## What version of Dapr?

0.11.3

## Expected Behavior

When the daprd docker image (daprio/daprd:0.11.3) is already on the machine, it is not pulled from the registry every time I deploy my application to k8s. 

## Actual Behavior

Every time I deploy my application to k8s, for every pod that is configured to be injected with a sidecar (using the Dapr annotations), the daprd Docker image is pulled from the registry.

This is (probably) because in the injector code, the `ImagePullPolicy` is set to `Allways`:
https://github.com/dapr/dapr/blob/d733ca0c50928c4ea80b85f763b6742e7498bb4f/pkg/injector/pod_patch.go#L456

This should be `IfNotPresent` i.m.o.

Here's the set of images I have on my machine:
```
> docker images
daprio/daprd        0.11.3              8ba6afd70647        2 weeks ago         81.6MB
daprio/daprd        0.11.2              d6552044b510        5 weeks ago         81.6MB
daprio/daprd        0.11.0              739548b95492        6 weeks ago         81.6MB
daprio/daprd        0.10.0              ee58a1db7312        2 months ago        76.1MB
daprio/daprd        0.8.0               bec8b93d7066        5 months ago        80.7MB
```

Here's the event log of one of the pods with sidecar when I deploy my application (taken from the k8s dashboard):

```
1. Successfully assigned eshop/ordering-6dfcc68f8d-5swfl to docker-desktop default-scheduler
2. Container image "eshop/ordering.api:linux-latest" already present on machine
3. Created container ordering-api
4. Started container ordering-api
5. Pulling image "docker.io/daprio/daprd:0.11.3"
6. Successfully pulled image "docker.io/daprio/daprd:0.11.3"
7. Created container daprd
8. Started container daprd
```

In line 5 you see that the image is pulled. It does this **every time I deploy the application** for **every pod** that is configured with a sidecar.

This makes that starting my application on k8s takes much longer than necessary because I have to wait for all the daprd images to be pulled down.

I use the stable Docker Desktop (on WSL2) v2.5.0.1 (49550)  with Kubernetes (v1.19.3) on Windows 10.

## Steps to Reproduce the Problem

1. Create a k8s manifest file for a service and annotate it with the Dapr annotations to inject a Dapr sidecar:

```yaml
kind: Deployment
...
  template:
    metadata:
      labels:
        service: ordering-api
      annotations:
        dapr.io/enabled: "true"
        dapr.io/app-id: "ordering-api"
        dapr.io/app-port: "80"
    spec:
      ...
```

2. Apply the manifest file to your k8s cluster:

```
kubectl apply -f ordering-api.yaml
```

3. Inspect the k8s event log (e.g. using the k8s dashboard).

## Release Note

RELEASE NOTE: **FIX** Changed ImagePullPolicy for Dapr sidecar Docker image from `Allways` to `IfNotPresent`

### PR DESCRIPTION
**Pull request by abserari**
# Description

We found that in injector, the imagePullPolicy config couldn't get effect to the container. So we add the feature for that.

## Issue reference

This will close #2444 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
Looks like end to end tests are failing

**Comment by abserari** [comment]
> Looks like end to end tests are failing

but I don't know if my fault.

Because in error log, it seems that e2e couldn't get redis start.

**Comment by yaron2** [comment]
> > Looks like end to end tests are failing
> 
> but I don't know if my fault.
> 
> Because in error log, it seems that e2e couldn't get redis start.
> 
> ```shell
> Run make setup-test-env-redis
> helm install dapr-redis stable/redis --wait --timeout 5m0s --namespace dapr-tests -f ./tests/config/redis_override.yaml
> Error: failed to download "stable/redis" (hint: running `helm repo update` may help)
> make: *** [setup-test-env-redis] Error 1
> tests/dapr_tests.mk:155: recipe for target 'setup-test-env-redis' failed
> Error: Process completed with exit code 2.
> ```

I'm rerunning the tests

**Comment by abserari** [comment]
> > > Looks like end to end tests are failing
> > 
> > 
> > but I don't know if my fault.
> > Because in error log, it seems that e2e couldn't get redis start.
> > ```shell
> > Run make setup-test-env-redis
> > helm install dapr-redis stable/redis --wait --timeout 5m0s --namespace dapr-tests -f ./tests/config/redis_override.yaml
> > Error: failed to download "stable/redis" (hint: running `helm repo update` may help)
> > make: *** [setup-test-env-redis] Error 1
> > tests/dapr_tests.mk:155: recipe for target 'setup-test-env-redis' failed
> > Error: Process completed with exit code 2.
> > ```
> 
> I'm rerunning the tests

Again, and same error log

**Comment by yaron2** [comment]
@youngbupark @tcnghia can you please take a look?

**Comment by EdwinVW** [comment]
This PR fixes issue #2444.

File: pkg/injector/pod_patch.go [hunk 1]
```diff
@@ -431,7 +431,20 @@ func getKubernetesDNS(name, namespace string) string {
...
+func getPullPolicy(pullPolicy string) corev1.PullPolicy {
+	switch pullPolicy {
+	case "Always":
+		return corev1.PullAlways
+	case "Never":
+		return corev1.PullNever
+	case "IfNotPresent":
+		return corev1.PullIfNotPresent
+	default:
+		return corev1.PullAlways
```
Comment (yaron2) [inline pkg/injector/pod_patch.go:443]: Can you change the default to be `PullIfNotPresent `?

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/injector/pod_patch.go (hunk 1 above)
Comment (abserari) [reply pkg/injector/pod_patch.go:443]: Ok

**Review by yaron2 — APPROVED** [review]
LGTM

**Comment by yaron2** [comment]
Thanks @abserari for this PR! that's a good use of the `SIDECAR_IMAGE_PULL_POLICY` environment variable we had and never used.

**Pull request merged**